src/js/stores/CosmosPackagesStore.js

```diff
@@ -299,6 +325,18 @@ class CosmosPackagesStore extends GetSetBaseStore {
     this.emit(COSMOS_DESCRIBE_ERROR, error, name, version);
   }
 
+  processPackageListVersionsSuccess(packageVersions) {
+    this.set({ packageVersions });
```

Comment:
I think we should store `packageVersions` by `name` because there is a chance that when `getPackageVersions` is called, that is returns versions for a different package. It would be nice if in `getPackageVersions` you pass in the package name and if the versions exist, no API request is made. Currently, if you click around and select different packages, a request will be made every time. The way we store `packageDetails` is also unfortunate because this also should be stored by name.

Reply:
Its addressed, some parts of the same work is to address was added to https://github.com/dcos/dcos-ui/pull/2397